bangazonapi/views/product.py

```diff
@@ -288,3 +297,43 @@ def min_price_filter(product):
         serializer = ProductSerializer(
             products, many=True, context={'request': request})
         return Response(serializer.data)
+
+    @action(methods=['post', 'delete'], detail=True)
+    def like(self, request, pk):
+        customer = Customer.objects.get(user=request.auth.user)
+
+        if request.method == "POST":
+            liked_product = LikeProduct()
+            liked_product.product = Product.objects.get(pk=pk)
+            liked_product.customer = customer
+            liked_product.save()
+
+            liked_product_json = LikedProductSerializer(liked_product, many=False, context={'request': request})
+
+            return Response(liked_product_json.data)
+
+        if request.method == "DELETE":
+            try:
+                liked_product = LikeProduct.objects.get(customer=customer, product__id=pk)
+                liked_product.delete()
+
+                return Response({}, status=status.HTTP_204_NO_CONTENT)
+
+            except LikeProduct.DoesNotExist as ex:
+                return Response({'message': ex.args[0]}, status=status.HTTP_404_NOT_FOUND)
+
+            except Exception as ex:
+                return Response({'message': ex.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+   
+   
+    @action(methods=['get'], detail=False)
+    def liked(self, request):
+        customer = Customer.objects.get(user=request.auth.user)
+   
+        liked = LikeProduct.objects.filter(customer=customer)
+
+        serializer = LikedProductSerializer(
+            liked, many=True, context={'request': request})
+        return Response(serializer.data)
```

Comment:
you have two `return` statements here, the latter of which will currently never execute.

try/except might be the way to go here, with one return in each.